### ISSUE TITLE
How to set the Dropout layer with random rate when training ?

### ISSUE THREAD

**Issue by Zhaosoi** [keras-team/keras#17242]
I want to change dropout rate when training.  The dropout rate is a random value between 0.1 and 0.9, and in every five times, dropout rate is set to 1. My code is as follows. But I found that the drorout rate  not changed with the change of training batch. Can anyone help me?

import numpy as np
import keras
from keras.models import Sequential, Model
from keras.wrappers.scikit_learn import KerasClassifier
from keras.utils import np_utils,plot_model
from sklearn.model_selection import cross_val_score,train_test_split,KFold
from sklearn.preprocessing import LabelEncoder
from keras.layers import Dense,Dropout,Flatten,Conv1D,MaxPooling1D,BatchNormalization,Input, Activation
import itertools
import tensorflow as tf
from tensorflow import keras
from tensorflow.keras import layers, Model
from keras.callbacks import TensorBoard
from keras.regularizers import l2
import math
from keras import backend as K
from keras.callbacks import Callback

**e=K.variable(np.random.uniform(0.1,0.9))**

# Define the Keras model to add callbacks to
def baseline_model():
    
    input = Input(shape = input_shape)
    print('input',input)
    print('aaaaa',e)
    xy = Dropout(**rate=e**, input_shape=input_shape)(input)
    print('bbbbbb',e)
    xy = Conv1D(filters=16, kernel_size=3, strides=1, input_shape=input_shape)(xy)
    xy = Conv1D(filters=16, kernel_size=3, activation='tanh')(xy)
    xy = MaxPooling1D(pool_size=3)(xy)
    xy = Flatten()(xy)
    output = Dense(units=9, activation='softmax')(xy)
    model = Model(inputs=input, outputs=output)
    model.compile(loss='categorical_crossentropy',optimizer='adam', metrics=['accuracy'])
    return model

# Load example MNIST data and pre-process it
(x_train, y_train), (x_test, y_test) = tf.keras.datasets.mnist.load_data()
x_train = x_train.reshape(-1, 784).astype("float32") / 255.0
x_test = x_test.reshape(-1, 784).astype("float32") / 255.0

# Limit the data to 1000 samples
x_train = x_train[:1000]
y_train = y_train[:1000]
x_test = x_test[:1000]
y_test = y_test[:1000]

class CustomCallback(keras.callbacks.Callback):
    
    **def __init__(self, e):
         self.e = = K.variable(np.random.uniform(0.1,0.9))**
    
    def on_train_begin(self, logs=None):
         keys = list(logs.keys())
         print("Starting training; got log keys: {}".format(keys))

    def on_train_end(self, logs=None):
        keys = list(logs.keys())
        print("Stop training; got log keys: {}".format(keys))

    def on_epoch_begin(self, epoch, logs=None):
        keys = list(logs.keys())
        print("Start epoch {} of training; got log keys: {}".format(epoch, keys))

    def on_epoch_end(self, epoch, logs=None):
         keys = list(logs.keys())
         print("End epoch {} of training; got log keys: {}".format(epoch, keys))

    **def on_train_batch_begin(self, batch, logs=None):
         keys = list(logs.keys())
         print("...Training: start of batch {}; got log keys: {}".format(batch, keys))
         self.e = K.variable(np.random.uniform(0.1,0.9))
         print('1111111111',batch)
         print('2222222222',self.e)
         if batch%5== 0 and batch!=0: # in every five times, dropout rate is set to 1 
             K.set_value(self.e, 1)
             print('33333333',self.e)**

    def on_train_batch_end(self, batch, logs=None):
        keys = list(logs.keys())
        print("...Training: end of batch {}; got log keys: {}".format(batch, keys))

model = baseline_model()
model.fit(
    x_train,
    y_train,
    batch_size=128,
    epochs=1,
    verbose=0,
    validation_split=0,
    callbacks=[CustomCallback(e)],
)

**Comment by tilakrayal** [comment]
@Zhaosoi,
Code shared is full of [indentation errors](https://colab.research.google.com/gist/tilakrayal/a56f3ef87166ad99f89336264951b637/untitled763.ipynb), please share a colab gist with issue reported or simple stand alone indented code with all dependencies such that we can replicate the issue reported. Thank you!

**Comment by Zhaosoi** [comment]
> @Zhaosoi, Code shared is full of [indentation errors](https://colab.research.google.com/gist/tilakrayal/a56f3ef87166ad99f89336264951b637/untitled763.ipynb), please share a colab gist with issue reported or simple stand alone indented code with all dependencies such that we can replicate the issue reported. Thank you!

Many thanks! I have uploaded the data (.csv) and code (.ipynb and .py)
[code.zip](https://github.com/keras-team/keras/files/10019346/code.zip)
)

[shuju.csv](https://github.com/keras-team/keras/files/10019323/shuju.csv)

**Comment by Zhaosoi** [comment]
> 



> @Zhaosoi, Code shared is full of [indentation errors](https://colab.research.google.com/gist/tilakrayal/a56f3ef87166ad99f89336264951b637/untitled763.ipynb), please share a colab gist with issue reported or simple stand alone indented code with all dependencies such that we can replicate the issue reported. Thank you!

@tilakrayal Many thanks! I have uploaded the data (.csv) and code (.ipynb and .py).

**Comment by tilakrayal** [comment]
@Zhaosoi,
I was not able to find any issue while executing the given code. Kindly find the gist of it [here](https://colab.research.google.com/gist/tilakrayal/459e54d2885c52c6078f7cf7f168d57a/crackmain5.ipynb) and please let us know if anything is missing here.
